src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -1054,6 +1057,96 @@ public void onNewRMarkdownDoc()
+            new ServerRequestCallback<JsObject>()
+            {
+               @Override
+               public void onResponseReceived(JsObject createdFiles)
+               {
+                  // Loop once to collect failures
+                  List<String> failedPaths = new ArrayList<String>();
+                  for (String filePath : JsUtil.asIterable(createdFiles.keys()))
+                  {
+                     boolean success = createdFiles.getBoolean(filePath);
+                     if (!success)
+                        failedPaths.add(filePath);
+                  }
+                  
+                  if (!failedPaths.isEmpty())
+                  {
+                     String message = "Failed to create the following files:\n";
+                     for (String path : failedPaths)
+                        message += "\n\t" + path;
+                     globalDisplay_.showErrorMessage("Error Creating Shiny Application", message);
+                     return;
+                  }
+                  
+                  // Open and focus files that we created
+                  for (String filePath : JsUtil.asIterable(createdFiles.keys()))
+                  {
+                     FileSystemItem path = FileSystemItem.createFile(filePath);
+                     openFile(path, FileTypeRegistry.R, new CommandWithArg<EditingTarget>()
+                     {
+                        @Override
+                        public void execute(EditingTarget target)
+                        {
+                           target.setCursorPosition(Position.create(1, 0));
+                        }
+                     });
+                  }
+               }
+               
+               @Override
+               public void onError(ServerError error)
+               {
+                  Debug.logError(error);
+               }
+            });
+   }
+   
+   @Handler
+   public void onNewRShinyApp()
+   {
+      dependencyManager_.withShiny("Creating Shiny applications", new Command()
+      {
+         @Override
+         public void execute()
+         {
+            server_.getWorkingDirectory(new ServerRequestCallback<String>()
+            {
+               @Override
+               public void onResponseReceived(String workingDir)
+               {
+                  NewShinyWebApplication widget = new NewShinyWebApplication(
+                        "New Shiny Web Application",
+                        workingDir,
+                        new OperationWithInput<NewShinyWebApplication.Result>()
+                        {
+                           @Override
+                           public void execute(Result input)
+                           {
+                              doNewRShinyApp(input);
+                           }
+                        });
+
+                  widget.showModal();
+               }
+
+               @Override
+               public void onError(ServerError error)
```

Comment:
I think you want to actually show there error to the user here (otherwise there will be a silent failure with no feedback). You can use the SimpleRequestCallback<T> class to automatically report the error via a standard onError handler.